libr/io/section.c

```diff
@@ -57,7 +57,10 @@ static RIOSection *findMatching (RIO *io, ut64 paddr, ut64 vaddr, ut64 size, ut6
 R_API RIOSection *r_io_section_add(RIO *io, ut64 offset, ut64 vaddr, ut64 size, ut64 vsize, int rwx, const char *name, ut32 bin_id, int fd) {
 	int update = 0;
 	RIOSection *s;
-	if (size == 0 || size > 0xf0000000) {
+	ut64 curr = lseek (fd, 0, SEEK_CUR);
```

Comment:
ok, the user shall define what fd is currently raised, which is represented on the one side by io->raised and on the other side by io->desc. If you want to know the size of a file, and use r_io_size, you have to ensure that io->desc is the same desc, when you are done with you stuff as it was before. In order to use r_io_size you must set io->desc to the desc of the file of what you want to know the size. So later you have to set io->desc back. And that is what r_io_desc_size does for you